### PR TITLE
11674: support for mapping an XAD to Address

### PR DESCRIPTION
**Pull request by mkalish**

### CODE REVIEW

File: src/main/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtils.java [hunk 1]
```diff
@@ -401,6 +421,19 @@ public static String getAddressType(String xad7Type, String xad18Type) {
         return addressType;
     }
 
+    // Follows the mapping here: https://docs.google.com/spreadsheets/d/1NwT6PQDhxnzaKcGOdZkioyHlt6WgaZBe_0KM_UJFNxE/edit#gid=0
+    public static String getFhirAddressType(String xad7Type) {
+        LOGGER.info("Calculating Address Type");
+        LOGGER.debug("Calculating address Type from XAD.7 {}", xad7Type);
```
Comment (jimmyfagan) [inline src/main/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtils.java:427]: How is this address type different than the `getAddressType` above? Can that difference be highlighted in these log messages?

File: src/test/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtilsTest.java [hunk 2]
```diff
@@ -269,6 +269,14 @@ void test_getAddressUse2() {
...
         assertThat(Hl7RelatedGeneralUtils.getAddressUse(null, null, null)).isEmpty();
 
+        // This uses the function that maps to the updated address use mappings
+        // https://docs.google.com/spreadsheets/d/1eJAihkIzcTOd9SY-xoXaboRDdUM6T8Wabi73wurU1hA/edit#gid=0
+        assertThat(Hl7RelatedGeneralUtils.getFhirAddressUse("BA")).isEqualTo("old");
+        assertThat(Hl7RelatedGeneralUtils.getFhirAddressUse("BI")).isEqualTo("billing");
+        assertThat(Hl7RelatedGeneralUtils.getFhirAddressUse("C")).isEqualTo("temp");
+        assertThat(Hl7RelatedGeneralUtils.getFhirAddressUse("B")).isEqualTo("work");
+        assertThat(Hl7RelatedGeneralUtils.getFhirAddressUse("H")).isEqualTo("home");
+        assertThat(Hl7RelatedGeneralUtils.getFhirAddressUse("O")).isEqualTo("work");
```
Comment (jimmyfagan) [inline src/test/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtilsTest.java:279]: Maybe add a case for unexpected xad7 values to confirm the empty string returned?

File: src/test/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtilsTest.java [hunk 3]
```diff
@@ -290,6 +298,11 @@ void test_getAddressType() {
         assertThat(Hl7RelatedGeneralUtils.getAddressType(null, ANYTHING)).isEmpty();
         assertThat(Hl7RelatedGeneralUtils.getAddressType(null, null)).isEmpty();
 
+        // This uses the function that maps to the updated address type mappings here
+        // https://docs.google.com/spreadsheets/d/1NwT6PQDhxnzaKcGOdZkioyHlt6WgaZBe_0KM_UJFNxE/edit#gid=0
+        assertThat(Hl7RelatedGeneralUtils.getFhirAddressType("M")).isEqualTo("postal");
+        assertThat(Hl7RelatedGeneralUtils.getFhirAddressType("SH")).isEqualTo("postal");
+
```
Comment (jimmyfagan) [inline src/test/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtilsTest.java:307]: Maybe add a case for unexpected xad7 values to confirm the empty string returned?

**Review by jimmyfagan — COMMENTED** [review]
Looks good, just very minor comments that will likely be quick fixes.

File: src/main/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtils.java (hunk 1 above)
Comment (mkalish) [reply src/main/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtils.java:427]: So this one maps to the current FHIR mappings from the spreadsheet vs the current one which does not map.

File: src/test/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtilsTest.java (hunk 2 above)
Comment (mkalish) [reply src/test/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtilsTest.java:279]: Will do

File: src/test/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtilsTest.java (hunk 3 above)
Comment (mkalish) [reply src/test/java/io/github/linuxforhealth/hl7/data/Hl7RelatedGeneralUtilsTest.java:307]: Will do

**Pull request merged**